src/tests/system/tests/test_identity.py

```diff
@@ -496,3 +496,44 @@ def test_identity__lookup_users_fully_qualified_name_and_case_insensitive(client
         result = client.tools.id(name)
         assert result is not None, f"User {name} was not found using id"
         assert result.memberof([103, 1003]), f"User {name} is member of wrong groups"
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.authentication
+@pytest.mark.topology(KnownTopologyGroup.AnyAD)
+def test_identity__lookup_idmapping_of_posix_and_non_posix_user_and_group(client: Client, provider: GenericADProvider):
+    """
+    :title: Check ID mapping of POSIX and non POSIX users in AD type directories when ldap_id_mapping is false
+    :setup:
+        1. Configure AD provider with ldap idmapping disabled
+    :steps:
+        1. Create user with POSIX attriubtes
+        2. Create group with POSIX attributes
+        3. Create user with no POSIX attributes
+        4. Create group with no POSIX attributes
+        5. Configure SSSD with "ldap_id_mapping" = false
+        6. Start SSSD
+    :expectedresults:
+        1. POSIX group information should be returned
+        2. POSIX user information should be returned
+        3. Non-POSIX group information should not be returned
+        1. Non-POSIX user information should not be returned
+    :customerscenario: False
+    """
+
+    u1 = provider.user("posix_user").add(uid=10001, gid=20001, password="Secret123",
+                                         gecos='User for tests',
+                                         shell='/bin/bash')
+    provider.group("posix_group").add(gid=20001).add_member(u1)
+
+    u2 = provider.user("nonposix_user").add(password="Secret123")
+    provider.group("nonposix_group").add().add_member(u2)
+
+    client.sssd.domain["ldap_id_mapping"] = "false"
+    client.sssd.start()
+
+    assert client.tools.getent.group("posix_group"), 'posix-group is not returned by sssd'
```

Comment:
We should check that the uid/gid matches the expected ones from setup.

Reply:
updated.

Reply:
@jakub-vavra-cz 

That's a good idea, ensure it's not pulling the identity from another source.